test/helpers/foundry/helpers.mts

```diff
@@ -0,0 +1,522 @@
+import { join, basename, extname, relative } from 'node:path';
+import { argv, stdout } from 'node:process';
+import { arch, platform } from 'node:os';
+import { Stream } from 'node:stream';
+import { pipeline } from 'node:stream/promises';
+import { createHash } from 'node:crypto';
+import { extract as extractTar } from 'tar';
+import { Open as Unzip, type Source, type Entry } from 'unzipper';
+import yargs from 'yargs/yargs';
+import { Minipass } from 'minipass';
+import {
+  type BinariesTuple,
+  type Checksums,
+  type PlatformArchChecksums,
+  Architecture,
+  Extension,
+  Binary,
+  Platform,
+  ParsedOptions,
+  ArchitecturesTuple,
+  PlatformsTuple,
+  DownloadOptions,
+} from './types.mts';
+
+export function noop() {}
+
+const Binaries = Object.values(Binary) as BinariesTuple;
+
+export function parseArgs(args: string[] = argv.slice(2)) {
+  const { $0, _, ...parsed } = yargs()
+    // Ensure unrecognized commands/options are reported as errors.
+    .strict()
+    // disable yargs's version, as it doesn't make sense here
+    .version(false)
+    // use the scriptName in `--help` output
+    .scriptName('yarn foundryup')
+    // wrap output at a maximum of 120 characters or `stdout.columns`
+    .wrap(Math.min(120, stdout.columns))
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true,
+    })
+    // enable ENV parsing, which allows the user to specify foundryup options
+    // via environment variables prefixed with `FOUNDRYUP_`
+    .env('FOUNDRYUP')
+    .command(['$0', 'install'], 'Install foundry binaries', (yargs) => {
+      yargs.options(getOptions()).pkgConf('foundryup');
+    })
+    .command('cache', '', (yargs) => {
+      yargs.command('clean', 'Remove the shared cache files').demandCommand();
+    })
+    .parseSync(args);
+
+  const command = _.join(' ');
+  switch (command) {
+    case 'cache clean':
+      return {
+        command,
+      } as const;
+    case '':
+    case 'install':
+      return {
+        command: 'install',
+        options: parsed as ParsedOptions<ReturnType<typeof getOptions>>,
+      } as const;
+  }
+  throw new Error(`Unknown command: '${command}'`);
+}
+
+function getOptions(
+  defaultPlatform = platform(),
+  defaultArch = normalizeSystemArchitecture(),
+) {
+  return {
+    binaries: {
+      alias: 'b',
+      type: 'array' as const,
+      multiple: true,
+      description: 'Specify the binaries to install',
+      default: Binaries,
+      choices: Binaries,
+      coerce: (values: Binary[]): Binary[] => [...new Set(values)], // Remove duplicates
+    },
+    checksums: {
+      alias: 'c',
+      description: 'JSON object containing checksums for the binaries.',
+      coerce: (rawChecksums: string | Checksums): Checksums => {
+        try {
+          return typeof rawChecksums === 'string'
+            ? JSON.parse(rawChecksums)
+            : rawChecksums;
+        } catch {
+          throw new Error('Invalid checksums');
```

Comment:
we could do a lot more validation of the checksum object... but I think it's bit premature, since maybe the way I implemented it is horirble.